### PR TITLE
EDUCATOR-4980

### PR DESCRIPTION
**Pull request by Andytr1**
Create team_workflow_mixin to handle calls to the team_workflow_api being created as part of [create team assessment api](https://github.com/edx/edx-ora2/pull/1383). 
 
[Jira](https://openedx.atlassian.net/browse/EDUCATOR-4980)

@edx/masters-devs-gta 

- [ ] Make sure your PR updates the version number in ``setup.py`` and ``package.json``
- [ ] Run ``make javascript sass`` if your changes include JS/CSS changes.
- [ ] Get a green Travis build for this PR
- [ ] Merge to master
- [ ] Create a release tag on GitHub https://github.com/edx/edx-ora2/releases
- [ ] Manually inspect diff at https://github.com/edx/edx-ora2/compare/0.x.n-1...0.x.n, email contributors
- [ ] Create PR to update `requirements/edx/{github.in,base.txt,development.txt,testing.txt}` in `edx-platform`.
- [ ] If manual testing of the changes against edx-platform is desired, create a sandbox (or use devstack).
- [ ] Get green build on the edx-platform PR, merge.
- [ ] Consider any feature flags that must be changed.
- [ ] Once your code has been released to production, try to test it there, too.

### CODE REVIEW

File: openassessment/xblock/team_workflow_mixin.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                Defaults to the submission created by the current student.
+
+        Returns:
+            dict
+
+        Raises:
+            AssessmentWorkflowError
+        """
+        if team_submission_uuid is None:
+            team_submission_uuid = self.get_submission_uuid()
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/team_workflow_mixin.py:63]: Good point. The api methods needed for teams will be done as part of https://openedx.atlassian.net/browse/EDUCATOR-4983 . At which point we should update workflow_mixin_get_submission_uuid. I am going to clarify it with a TODO.

[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/team_workflow_mixin.py:63]: Good catch. Fixed.

File: openassessment/xblock/team_workflow_mixin.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        if self.submission_uuid is not None:
+            return self.submission_uuid
+
+        elif self.teams_enabled:
+            from submissions.api import get_submissions, SubmissionInternalError
+
+            try:
+                # Query for submissions by the student item
+                student_item = self.get_student_item_dict()
+                submission_list = get_submissions(student_item)
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/team_workflow_mixin.py:95]: I have updated a TODO. I am not aware of a way to stub a an API method that doesn't exist yet.

File: openassessment/xblock/test/test_team_workflow_mixin.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Class to test the API between team_workflow_mixin and the team_workflow_api
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.team_mixin = TeamWorkflowMixin()
+
+    @patch('openassessment.xblock.team_workflow_mixin.team_workflow_api')
+    def test_get_team_workflow_info(self, api_mock):
+        submission_uid = "sub_uid"
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/test/test_team_workflow_mixin.py:22]: Fixed.

File: openassessment/xblock/team_workflow_mixin.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+from openassessment.workflow import team_api as team_workflow_api
+
+
+class TeamWorkflowMixin:
+    """
+    Handle OpenAssessment, team centric XBlock requests to the Workflow API.
+    """
+
+    @XBlock.json_handler
+    def handle_workflow_info(self, data, suffix=''):    # pylint:disable=W0613
```
Comment (jansenk) [inline openassessment/xblock/team_workflow_mixin.py:17]: I believe this should be handle_team_workflow_info

File: openassessment/xblock/team_workflow_mixin.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jansenk) [reply openassessment/xblock/team_workflow_mixin.py:91]: I'd suggest creating a stub function (just return none) and mock the behavior away in the tests

**Comment by Andytr1** [comment]
> Looks okay, but do we have to worry about `update_workflow_status`? I know the teams ORAs only have one step (staff), but based on the comments, is it called when all steps have been completed?

I don't think we have to worry (implement) `update_workflow_status`. Based on codescans, `update_workflow_status` is called from student/peer/self_assessment_mixins, but is **NOT** called from staff_assessment_mixin. Since team assessment is fairly identical to staff, I don't think this method is needed.

**Pull request merged**